extensions/binding/org.eclipse.smarthome.binding.bluetooth/src/main/java/org/eclipse/smarthome/binding/bluetooth/notification/BluetoothScanNotification.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.bluetooth.notification;
+
+/**
+ * The {@link BluetoothScanNotification} provides a notification of a received scan packet
+ *
+ * @author Chris Jackson - Initial contribution
+ */
+public class BluetoothScanNotification extends BluetoothNotification {
+    /**
+     * The receive signal strength for this beacon packet
+     */
+    private int rssi = Integer.MIN_VALUE;
+
+    /**
+     * The raw data
+     */
+    private int[] data = null;
+
+    /**
+     * The manufacturer specific data
+     */
+    private int[] manufacturerData = null;
+
+    /**
+     * The beacon type
+     */
+    private BluetoothBeaconType beaconType = BluetoothBeaconType.BEACON_UNKNOWN;
+
+    /**
+     * The device name
+     */
+    private String name = new String();
+
+    /**
+     * An enumeration of basic beacon types
+     */
+    public enum BluetoothBeaconType {
+        BEACON_UNKNOWN,
+        BEACON_ADVERTISEMENT,
+        BEACON_SCANRESPONSE
+    }
+
+    /**
+     * Sets the receive signal strength RSSI value for the scan
+     *
+     * param rssi the RSSI value for the scan packet in dBm
+     */
+    public void setRssi(int rssi) {
+        this.rssi = rssi;
+    }
+
+    /**
+     * Gets the receive signal strength RSSI value for the scan
+     *
+     * @return the RSSI value for the scan packet in dBm or Integer.MIN_VALUE if no RSSI is available.
+     */
+    public int getRssi() {
+        return rssi;
+    }
+
+    /**
+     * Sets the scan packet data
+     *
+     * @param data an int array containing the raw packet data;
+     */
+    public void setData(int[] data) {
```

Comment:
do we really want to have arrays in our API?

Reply:
Well, those is the raw data of telegrams - so I would say that arrays are ok to use for this.
It is also [what the Android BLE API is doing](https://developer.android.com/reference/android/bluetooth/BluetoothAdapter.LeScanCallback.html#onLeScan(android.bluetooth.BluetoothDevice, int, byte[])).

The only thing I would claim should be changed is the int to byte - as the Bluetooth spec says that this content is all uInt8 - so just as Android does it, the best Java type seems to be byte (or was there any specific reason why you chose int, @cdjackson?).